recipes/recastnavigation/all/conanfile.py

```diff
@@ -26,36 +26,37 @@ class RecastNavigationConan(ConanFile):
 
     short_paths = True
 
-    def export_sources(self):
-        export_conandata_patches(self)
-
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            try:
-                del self.options.fPIC
-            except Exception:
-                pass
+            self.options.rm_safe("fPIC")
 
     def layout(self):
         cmake_layout(self, src_folder="src")
 
     def source(self):
-        get(self, **self.conan_data["sources"][self.version],
-            destination=self.source_folder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
     def generate(self):
         tc = CMakeToolchain(self)
-        tc.variables["RECASTNAVIGATION_DEMO"] = False
-        tc.variables["RECASTNAVIGATION_TESTS"] = False
-        tc.variables["RECASTNAVIGATION_EXAMPLES"] = False
+        tc.cache_variables["RECASTNAVIGATION_DEMO"] = False
+        tc.cache_variables["RECASTNAVIGATION_TESTS"] = False
+        tc.cache_variables["RECASTNAVIGATION_EXAMPLES"] = False
+        tc.cache_variables["RECASTNAVIGATION_STATIC"] = not self.options.shared
+        tc.variables["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = self.options.shared
         tc.generate()
 
+    def _patch_sources(self):
+        if self.version == "cci.20200511":
+            replace_in_file(self, os.path.join(self.source_folder, "CMakeLists.txt"),
+                            "project(RecastNavigation)",
+                            "project(RecastNavigation)\ninclude(GNUInstallDirs)")
```

Comment:
Would be better placed in a patch to keep recipe code as simple as possible.

Reply:
I agree, I'll move it to a patch after the build is finished to see whether `include(GNUInstallDirs)` is sufficient for correct DLL output (it should be).

Reply:
It won't be sufficient